internal/data/disbursement_instructions.go

```diff
@@ -95,15 +95,18 @@ func (di DisbursementInstructionModel) ProcessAll(ctx context.Context, opts Disb
 	// We need all the following logic to be executed in one transaction.
 	return db.RunInTransaction(ctx, di.dbConnectionPool, nil, func(dbTx db.DBTransaction) error {
 		// Step 1: Fetch all receivers by contact information (phone, email, etc.) and create missing ones
-		receiversByIDMap, err := di.reconcileExistingReceiversWithInstructions(ctx, dbTx, opts.Instructions, opts.ReceiverContactType)
+		registrationContactType := opts.Disbursement.RegistrationContactType
+		receiversByIDMap, err := di.reconcileExistingReceiversWithInstructions(ctx, dbTx, opts.Instructions, registrationContactType)
 		if err != nil {
 			return fmt.Errorf("processing receivers: %w", err)
 		}
 
 		// Step 2: Fetch all receiver verifications and create missing ones.
-		err = di.processReceiverVerifications(ctx, dbTx, receiversByIDMap, opts.Instructions, opts.Disbursement, opts.ReceiverContactType)
-		if err != nil {
-			return fmt.Errorf("processing receiver verifications: %w", err)
+		if !registrationContactType.IncludesWalletAddress {
+			err = di.processReceiverVerifications(ctx, dbTx, receiversByIDMap, opts.Instructions, opts.Disbursement, registrationContactType)
+			if err != nil {
+				return fmt.Errorf("processing receiver verifications: %w", err)
+			}
 		}
 
 		// Step 3: Fetch all receiver wallets and create missing ones
```

Comment:
The receiverWallet creation step gets tricky here... considering that we don't need to select a wallet application, do we actually need the wallet ID to be selected by the user?

💡  Maybe we should have a special wallet called something like Custom/Native/UserManaged or something of the sorts. We can filter it off from the results of GET /wallets in another ticket.